SocialCareCaseViewerApi.Tests/V1/UseCase/SubmissionUseCaseTests.cs

```diff
@@ -0,0 +1,90 @@
+            var worker = TestHelpers.CreateWorker();
+            var resident = TestHelpers.CreatePerson();
+
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(request.CreatedBy)).Returns(worker);
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(request.ResidentId)).Returns(resident);
+            _mockMongoGateway.Setup(x => x.InsertRecord(It.IsAny<string>(), It.IsAny<CaseSubmission>()));
+
+            var (caseSubmissionResponse, caseSubmission) = _submissionsUseCase.ExecutePost(request);
+            var expectedResponse = TestHelpers.CreateCaseSubmission(SubmissionState.InProgress,
+                caseSubmission.CreatedAt, worker, resident, caseSubmission.SubmissionId, request.FormId);
+
+            caseSubmissionResponse.Should().BeEquivalentTo(expectedResponse.ToDomain().ToResponse());
+            _mockDatabaseGateway.Verify(x => x.GetWorkerByEmail(request.CreatedBy), Times.Once);
+            _mockDatabaseGateway.Verify(x => x.GetPersonByMosaicId(request.ResidentId), Times.Once);
+            _mockMongoGateway.Verify(x => x.InsertRecord(CollectionName, caseSubmission), Times.Once);
+        }
+
+        [Test]
+        public void ExecutePostThrowsAnErrorWhenNoWorkerFoundForRequest()
+        {
+            var request = TestHelpers.CreateCaseSubmissionRequest();
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(request.CreatedBy));
+
+            Action act = () => _submissionsUseCase.ExecutePost(request);
+
+            act.Should().Throw<WorkerNotFoundException>()
+                .WithMessage($"Worker with email {request.CreatedBy} not found");
+        }
+
+        [Test]
+        public void ExecutePostThrowsAnErrorWhenNoResidentFoundForRequest()
+        {
+            var request = TestHelpers.CreateCaseSubmissionRequest();
+            var worker = TestHelpers.CreateWorker();
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(request.CreatedBy)).Returns(worker);
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(request.ResidentId));
+
+            Action act = () => _submissionsUseCase.ExecutePost(request);
+
+            act.Should().Throw<PersonNotFoundException>()
+                .WithMessage($"Person with id {request.ResidentId} not found");
+        }
+
+        [Test]
+        public void ExecuteGetByIdSucessfully()
+        {
+            var submissionResponse = TestHelpers.CreateCaseSubmission();
+            _mockMongoGateway.Setup(x => x.LoadRecordById<CaseSubmission>(It.IsAny<string>(), submissionResponse.SubmissionId)).Returns(submissionResponse); ;
+            var response = _submissionsUseCase.ExecuteGetById(submissionResponse.SubmissionId);
+            _mockMongoGateway.Verify(x => x.LoadRecordById<CaseSubmission>(It.IsAny<string>(), submissionResponse.SubmissionId), Times.Once);
+            response.Should().BeEquivalentTo(submissionResponse.ToDomain().ToResponse());
+        }
+
+
```

Comment:
Can we have a test where LoadRecordById returns null, we would expect our function to also return null in this instance